File: 875-koko-eating-bananas/875. Koko Eating Bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)
        while l < r:
            mid = (l + r) // 2
            hour_spent = 0
            for pile in piles:
                hour_spent += math.ceil(pile / mid)
            if hour_spent > h:
                l = mid + 1
            else:
                r = mid
        return l

File: 875-koko-eating-bananas/875. Koko Eating Bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        sort(piles.begin(), piles.end());
        int l = 1, r = piles[piles.size()-1];
        int ans = INT_MAX;
        while(l < r) {
            int k = (l + r) / 2;
            int time = 0;
            for(int i=0; i<piles.size(); i++) {
                time += (piles[i] + k - 1) / k;
            }
            if(time > h) {
                l = k + 1;
            } else {
                r = k;
            }
        }
        return r;
    }
};